src/Tools/dotnet-monitor/GenerateApiKeyCommandHandler.cs

```diff
@@ -21,24 +21,57 @@ namespace Microsoft.Diagnostics.Tools.Monitor
     internal sealed class GenerateApiKeyCommandHandler
     {
         public Task<int> GenerateApiKey(CancellationToken token, IConsole console)
+        {
+            GeneratedApiKey newKey = GeneratedApiKey.Create();
+
+            console.Out.WriteLine(FormattableString.Invariant($"Authorization: {Monitoring.RestServer.AuthConstants.ApiKeySchema} {newKey.MonitorApiKey}"));
+            console.Out.WriteLine(FormattableString.Invariant($"ApiKeyHash: {newKey.HashValue}"));
+            console.Out.WriteLine(FormattableString.Invariant($"ApiKeyHashType: {newKey.HashAlgorithm}"));
+
+            return Task.FromResult(0);
+        }
+    }
+
+    internal sealed class GeneratedApiKey
+    {
+        private const int KeyLengthBytes = 32;
```

Comment:
Because this class is parameterized to allow different hash algorithms, this shouldn't be a constant. I think you can use `hasher.HashSize / 8` instead so that if we use SHA512, it would automatically work.

Reply:
No, this is a hard coded constant at [ApiKeyAuthenticationHandler.cs:66](https://github.com/dotnet/dotnet-monitor/blob/c6b8b937134cc3b732f8a6766e846a6348f5e251/src/Tools/dotnet-monitor/Auth/ApiKeyAuthenticationHandler.cs#L66). 
It probably makes sense to have `ApiKeyAuthenticationHandler` reference this from `GeneratedApiKey` or vice versa and have this be a single parameter.

Reply:
Just because something else in dotnet-monitor is making a bad assumption doesn't mean you should code a different part to assume that same bad assumption.

The intent was to allow other algorithms to be use, which can have different key lengths. Hence why there is an option for specifying the algorithm used to hash the original key. Otherwise, we'd just bake it in to use SHA256.

Reply:
Just to be clear here, this value represents the length of the `MonitorApiKey` not the `HashValue` which would be determined by the algorithm, say `SHA512` vs `SHA256`. We assume the "private key" you'll use is always 32 bytes, and it just happens to be that we use SHA256 by default which also happens to provide a 32 byte output hash value.
There is no reason the `MonitorApiKey` always needs to be 32 bytes, but there also isn't a reason not to require it to be 32 bytes.

Reply:
Although I think you're onto something, I tried this as `SHA512` and it gets rejected.

Reply:
Ahh, I copy/pasted the `SHA256.Create()` without realizing it. I intended to make this `HashAlgorithm.Create(hashAlgorithm)`